pkg/ts/recipe/ParticleCook.ts

```diff
@@ -31,7 +31,10 @@ export class ParticleCook {
   }
   static specToMeta(spec) {
     // TODO(sjmiles): impedance mismatch here is likely to cause problems
-    const {$kind: kind, $container: container, $staticInputs: staticInputs} = spec;
+    const {$kind: kind, $container: container, $staticInputs: staticInputs, $bindings: bindings} = spec;
+    if (bindings) {
```

Comment:
Because this is not really part of the normal program flow, I would put this at the top of the function and test `spec.$bindings` directly. This whole function is probably a mistake ... the meta should just have the $.

Reply:
moved the if block.
re side note - i agree, it would be nice to have consistency for $ or no $... but probably best to change when we have more tests :)